airflow/providers/common/sql/hooks/sql.py

```diff
@@ -30,6 +30,25 @@
 from airflow.version import version
 
 
+def has_scalar_return_value(sql: str | Iterable[str], return_last: bool, split_statements: bool):
+    """
+    Determines when scalar value should be returned.
+
+    Scalar value should be returned when:
```

Comment:
I was thinking about it - I think that terminology is confusing a bit. When we think about `scalar` we think about one value. But execution of query always return an iterator of rows (IIRc). 

So at the end we have following possibilities:

* a list of query results (list of lists of rows) - when we have multiple queries
* a single query result (list of rows) - when we have one query or when we use `return_last`

maybe rename it to something like `single_result` or something like that ?

Reply:
I changed it everywhere to `return_single_query_results`. Note singular `query` but plural `results` :D